pucas/ldap.py

```diff
@@ -84,9 +84,30 @@ def user_info_from_ldap(user):
     user_info = LDAPSearch().find_user(user.username)
     if user_info:
         for user_attr, ldap_attr in attr_map.items():
-            # NOTE: is it possible some values are not set at all here,
-            # in which case we'd get 'None' string values set?
-            setattr(user, user_attr, str(getattr(user_info, ldap_attr)))
+            # Handle issues where an attribute may need to be populated by
+            # multiple attributes OR where it is missing, in which case
+            # handle the LDAPCursorError and set None to empty string.
+
+            # if the user passed a list, try in order until we get the attr
+            # or we fail to do so.
+            if isinstance(ldap_attr, list):
```

Comment:
There is logic duplicated in the if/else here that shouldn't be. What if you make single entries into a list of attributes first so you can always treat it as a list and set the first available attribute?

Reply:
I like that idea and will implement that. I don't foresee major overhead in a single item list vs. the cleaner code.